### DISCUSSION TITLE
Product and variant webhooks

### DISCUSSION THREAD

**Discussion by corygibbons** [swellstores/community#91] [Help] [answered]
Is there any documentation around what will or won't trigger product and variant webhooks, particularly around `.updated`?

If I have a product with variants then only change something about a variant (`status` or `sku`) from the product edit page (`/admin/products/{id}`) the `product.updated` webhook fires but `variant.updated` doesn't, but seems like it should?

If I open the modal to edit a variant and change the `status` or `sku` the `variant.updated` doesn't fire, but it will if I change the `price`. Seems that it should fire if anything on the variant changes?

**Comment by greph** [comment, marked as answer]
Hey @corygibbons , we have some documentation around the [webhook model](https://developers.swell.is/backend-api/webhooks/the-webhook-model), though nothing specifically referencing the `product.updated` event.

When saving information on the variant level, only the `variant.update` event should fire. The `product.update` event will fire however when saving the product page, which may be causing confusion (the page prompts you to save if you navigate away). One workaround, if you would only like the `variant.update` to fire, could be editing the variant only (More button).
